docs/proposals/multiple-sources-for-applications-ui.md

```diff
@@ -0,0 +1,225 @@
+currently shows one ApplicationSource, regardless of whether the application has one source or multiple sources. With the PR 10432 implementation, if the application has multiple sources,
+the UI displays only the first one from the list of sources. Also, in particular, in the Summary tab, the source parameters are non-editable. 
+
+2. History and Rollback is disabled for multi-source applications. The button is disabled. Jorge has submitted a PR for 
+rollback which includes controller and UI changes [here](https://github.com/argoproj/argo-cd/pull/14124).
+
+   <img height="40%" width="40%" src="images/history-and-rollback-button.png"/>
+
+3. The New Application dialog currently only allows users to provide one source.
+
+Thus, multiple source applications are not considered first class citizens in the UI.
+
+Note, see the Open Questions section for concerns regarding the priority or value of some of the above changes.
+
+## Motivation
+
+The motivation behind this change is to add a more complete story for the multiple source feature. The UI should support 
+the creation of multiple source applications, and also support the viewing and editing of parameters from all sources. The three
+points in the summary above are the base or core changes that need to be addressed.
+
+### Goals
+
+The goals of the proposal are:
+
+- Provide first-class support of multiple sources for applications in the UI (eg. address the aforementioned restrictions)
+- Outline stages of implementation that will help ease PR review, and reduce the risk of introducing regressions/issues.
+
+
+### Non-goals
+* The design changes for the Argo CD CLI is beyond the scope of this proposal (The server APIs can probably be reused)
+
+## Proposal
+
+As mentioned in the previous summary section, the application source parameters are surfaced in the UI in three locations.
+The Resource details pages, specifically, the Summary and Parameters tabs, the deployment history, and the Application
+Create panel page. These pages should be updated.
+
+### Resource Details
+
+The following describes the current behavior and proposed changes for the Summary tab and the Parameters Tab.
+
+#### i) Summary Tab
+
+_Current Behavior:_ 
+
+The current Summary tab includes source-related information, including the repository. For example, in Figure 1 below,
+the REPO URL and PATH.
+
+<img height="50%" width="50%" src="images/current-summary-tab.png"/>
+
+Figure 1: The current Summary tab
+
+_Proposed Change:_
+
+To support multiple sources, the source-related information, from a single-source-based design, will be ‘pulled out’ 
+and put into a new tab called **Sources**, and it will be combined with the **Parameters** tab (more details following). 
+The new **Sources** tab will allow users to view all the information related to each source, including the repo URL 
```

Comment:
I'd suggest `Inputs` over `Sources`. A `parameter` is an `input` and a `source` is an `input`, but a `parameter` is not a `source`.

Reply:
@jsoref , thanks for the feedback.  Good suggestion.  The reason I chose Sources is that it matches the new sources field in the application manifest. Perhaps we can finalize this label in the PR, especially we can see how it will look too with the other tabs in context.